internal/app/machined/pkg/controllers/network/device_config.go

```diff
@@ -156,13 +169,48 @@ func (ctrl *DeviceConfigController) Run(ctx context.Context, r controller.Runtim
 	}
 }
 
-func (ctrl *DeviceConfigController) getDeviceBySelector(device talosconfig.Device, links []resource.Resource) error {
+func (ctrl *DeviceConfigController) getDeviceBySelector(device talosconfig.Device, links safe.List[*network.LinkStatus]) error {
```

Comment:
Since are type asserting to `*v1alpha1.Device` anyway in `Run` maybe it makes sense to pass the concrete type here?

Reply:
thanks, fixed both of them